### PR TITLE
fix(plugin-dl): support empty definitions and next-line content

### PR DESCRIPTION
**Pull request by brc-dd**
Stacked on #600 — review the last commit only.

### Rationale

`markdown-it-deflist@4.0.0` allows empty definitions and definition content starting on the line after the marker (both Pandoc-documented). `@mdit/plugin-dl` still rejects a bare `:`/`~` marker, so such input renders as literal paragraph text — or, mid-list, the marker line and the following term get swallowed into the previous `<dd>` as lazy continuation, mangling the list structure for migrating users.

### Repro

```js
import MarkdownIt from "markdown-it";
import { dl } from "@mdit/plugin-dl";

const md = MarkdownIt({ linkify: true }).use(dl);
```

Input:

```markdown
Term
:
  Definition
```

Current `@mdit/plugin-dl` output:

```html
<p>Term
:
Definition</p>
```

markdown-it@14.3.0 + markdown-it-deflist@4.0.0 output (fixed output matches):

```html
<dl>
<dt>Term</dt>
<dd>Definition</dd>
</dl>
```

Mid-list input:

```markdown
Term
: def
:
Term2
: x
```

Current `@mdit/plugin-dl` output:

```html
<dl>
<dt>Term</dt>
<dd>def
:
Term2</dd>
<dd>x</dd>
</dl>
```

markdown-it@14.3.0 + markdown-it-deflist@4.0.0 output (fixed output matches):

```html
<dl>
<dt>Term</dt>
<dd>def</dd>
<dd></dd>
<dt>Term2</dt>
<dd>x</dd>
</dl>
```

### Root cause

`checkAndSkipMarker` in `packages/plugin-dl/src/plugin.ts` returns `-1` when nothing but whitespace follows the marker (`pos >= max`), and requires a space after the marker unconditionally.

### Fix

Port upstream's relaxed check: only require a space after the marker when inline content follows (`start < max && start === pos`). Existing tests pinning the old rejection behavior (bare-marker inputs in "should not render invalid definition lists" and the "Marker followed by only spaces" edge case) were updated to the upstream-parity outputs, verified verbatim against markdown-it-deflist@4.0.0. One docs sentence claiming a definition must be followed by paragraphs was corrected accordingly (en + zh).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/plugin-dl/__tests__/dl.spec.ts [hunk 1]
```diff
@@ -502,45 +544,82 @@ chili's
...
+: x
+`,
+        `\
+<dl>
+<dt>Term</dt>
+<dd>def</dd>
+<dd></dd>
+<dt>Term2</dt>
+<dd>x</dd>
+</dl>
```
Comment (Mister-Hope) [inline packages/plugin-dl/__tests__/dl.spec.ts:988]: Image

This differs with pandoc, please also check the others.

**Review by Mister-Hope — CHANGES_REQUESTED** [review]

**Comment by Mister-Hope** [comment]
Update: I think we do not need to 100% compact with PanDoc (we already choose not to add line breaks as prefix/suffix). An empty def is useless, and people can hack with `: &nbsp;` to achieve similar things.

I understand the pr goal is to align pandoc and upstream plugin behavior, but it is necessary for such an edge case like this?

**Comment by brc-dd** [comment]
I'm closing this for now. I'll check the dl and ruby plugins once again manually.

File: packages/plugin-dl/__tests__/dl.spec.ts (hunk 1 above)
Comment (brc-dd) [reply packages/plugin-dl/__tests__/dl.spec.ts:988]: convert from commonmark_x:

image

**Comment by brc-dd** [comment]
I've checked again and it seems like this is the only difference left between upstream (https://github.com/markdown-it/markdown-it-deflist/commit/d4c086ab02c9f63e2eadfc6b6ea0fc6956ce1b6e). The fix is practically one line change and can be added if there isn't any major issue with it.
